tf_encrypted/keras/engine/base_layer.py

```diff
@@ -88,3 +100,15 @@ def __call__(self, inputs, *args, **kargs):
   @property
   def prot(self):
     return get_protocol()
+
+  @property
+  def name(self):
+    return self._name
+
+  def _init_set_name(self, name, zero_based=True):
```

Comment:
is this normal behaviour? wouldn't users expect layers to have a name close to what they specified? (ie perhaps with an added counter but that's it)

Reply:
this is normal behavior (copied nearly verbatim from the keras version of this class or a superclass). the default name is to take the class name and snake case it, but if a user specifies their own name to the constructor it will be used instead (see where this gets called in the constructor for specifics)